From 1eeb196b3315447fe81b2aa5a916f791e32b4fd5 Mon Sep 17 00:00:00 2001
From: Alex Zhuravlev <alexey.zhuravlev@intel.com>
Date: Tue, 16 Apr 2013 19:21:13 +0400
Subject: [PATCH 13/13] LU-3179 fids: fix compilation error with gcc 4.7.2

initialize oi.oi.oi_id which gcc 4.7.2 is afraid of being
used later.

Signed-off-by: Alex Zhuravlev <alexey.zhuravlev@intel.com>
Change-Id: Iaeb6eac01340d80786463efe67ff74479017f074
---
 lustre/include/lustre_fid.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lustre/include/lustre_fid.h b/lustre/include/lustre_fid.h
index 1b6d0ff..1c37c7a 100644
--- a/lustre/include/lustre_fid.h
+++ b/lustre/include/lustre_fid.h
@@ -624,6 +624,7 @@ static inline void ost_fid_build_resid(const struct lu_fid *fid,
 {
 	if (fid_is_mdt0(fid) || fid_is_idif(fid)) {
 		struct ost_id oi;
+		oi.oi.oi_id = 0; /* gcc 4.7.2 complains otherwise */
 		if (fid_to_ostid(fid, &oi) != 0)
 			return;
 		ostid_build_res_name(&oi, resname);
-- 
1.8.2.1

